### PR TITLE
sd-bus: Small cleanups

### PR DESCRIPTION
**Pull request by namhyung**
Very simple code clean-ups using safe_close() and mfree().

### CODE REVIEW

File: src/libsystemd/sd-bus/sd-bus.c [hunk 1]
```diff
@@ -73,13 +73,8 @@ static void bus_close_fds(sd_bus *b) {
...
 
-        if (b->input_fd >= 0)
-                safe_close(b->input_fd);
-
-        if (b->output_fd >= 0 && b->output_fd != b->input_fd)
-                safe_close(b->output_fd);
-
-        b->input_fd = b->output_fd = -1;
+        b->input_fd = safe_close(b->input_fd);
+        b->output_fd = safe_close(b->output_fd);
```
Comment (dvdhrm) [inline src/libsystemd/sd-bus/sd-bus.c:77]: This fails if `input_fd == output_fd`, as you might have noticed in the CI test (or simply run ./test-gvariant locally and see the double-close()). But you could still optimize this to:

``` c
if (b->output_fd != b->input_fd)
        safe_close(b->output_fd);
b->output_fd = b->input_fd = safe_close(b->input_fd);
```

**Comment by dvdhrm** [comment]
Apart from the double-close, looks good!

**Comment by dvdhrm** [comment]
Please close this and send v2 as a new PR.

File: src/libsystemd/sd-bus/sd-bus.c (hunk 1 above)
Comment (namhyung) [inline src/libsystemd/sd-bus/sd-bus.c:77]: Ah, right.  I'll change it like above.  Thanks!

**Comment by namhyung** [comment]
OK, will send v2 soon.

**Pull request closed without merging**